cmd/src/args/topic.rs

```diff
@@ -0,0 +1,70 @@
+use crate::args::common::ListMode;
+use clap::{Args, Subcommand};
+
+#[derive(Debug, Subcommand)]
+pub(crate) enum TopicAction {
+    /// Create topic with given ID, name, number of partitions
+    /// and expiry time for given stream ID
+    Create(TopicCreateArgs),
+    /// Delete topic with given ID in given stream ID
+    Delete(TopicDeleteArgs),
+    /// Update topic name an message expiry time for given topic ID in given stream ID
+    Update(TopicUpdateArgs),
+    /// Get topic detail for given topic ID and stream ID
+    Get(TopicGetArgs),
+    /// List all topics in given stream ID
+    List(TopicListArgs),
+}
+
+#[derive(Debug, Args)]
+pub(crate) struct TopicCreateArgs {
+    /// Stream ID to create topic
+    pub(crate) stream_id: u32,
+    /// Topic ID to create
+    pub(crate) topic_id: u32,
+    /// Number of partitions inside the topic
+    pub(crate) partitions_count: u32,
+    /// Name of the topic
+    pub(crate) name: String,
+    /// Message expiry time in seconds
+    pub(crate) message_expiry: Option<u32>,
+}
+
+#[derive(Debug, Args)]
+pub(crate) struct TopicDeleteArgs {
+    /// Stream ID to delete topic
+    pub(crate) stream_id: u32,
+    /// Topic ID to delete
+    pub(crate) topic_id: u32,
+}
+
+#[derive(Debug, Args)]
+pub(crate) struct TopicUpdateArgs {
+    /// Stream ID to update topic
+    pub(crate) stream_id: u32,
+    /// Topic ID to update
+    pub(crate) topic_id: u32,
+    /// New name for the topic
+    pub(crate) name: String,
+    /// New message expiry time in seconds
+    /// (skipping parameter causes removal of expiry parameter in topic)
+    pub(crate) message_expiry: Option<u32>,
```

Comment:
Just a simple idea - maybe we could provide a string, e.g. "30s", "10h", "5d" etc. ("" or "none" for the disabled expiry?) and then just send the appropriate amount of the seconds to the server?

Reply:
Like enum option limited to only given values?

Like this:
```
Update topic name an message expiry time for given topic ID in given stream ID

Usage: iggy topic update <STREAM_ID> <TOPIC_ID> <NAME> [MESSAGE_EXPIRY]

Arguments:
  <STREAM_ID>       Stream ID to update topic
  <TOPIC_ID>        Topic ID to update
  <NAME>            New name for the topic
  [MESSAGE_EXPIRY]  New message expiry time in seconds (skipping parameter causes removal of expiry parameter in topic) [possible values: 30s, 10h, 5d, none]

Options:
  -h, --help  Print help

```

Reply:
Value could be anything (time + unit), I thought of maybe using something like [this](https://github.com/uutils/parse_datetime) and allowing the user to specify an arbitrary string value, which then would be parsed to the proper duration and sent further to the server (for `None` / 0 expiry, we could reserve a special values like an empty string, "none" etc.)

Reply:
with new commit one can call the `iggy` tool as following

`$ topic create 1 7 2 topic-name 3m 2y 23days 3sec 1minute`